src/main/java/zeenea/connector/Item.java

```diff
@@ -0,0 +1,183 @@
+package zeenea.connector;
+
+import static zeenea.connector.exception.ExceptionUtils.throwIfNull;
+
+import java.math.BigDecimal;
+import java.net.URI;
+import java.time.Instant;
+import java.util.*;
+import org.jetbrains.annotations.NotNull;
+import zeenea.connector.common.ItemIdentifier;
+import zeenea.connector.contact.ContactRelation;
+import zeenea.connector.property.*;
+
+/**
+ * Base class for all kinds of source items. An item is something that is managed and searchable in
+ * Zeenea Data Catalog.
+ *
+ * @since 1.0.0
+ */
+public abstract class Item {
+
+  private final String name;
+
+  private final ItemIdentifier id;
+
+  private final String description;
+
+  private final Map<String, PropertyValue> properties;
+
+  private final Instant updateTime;
+
+  private final Collection<ContactRelation> contactRelations;
+
+  protected Item(Builder<?, ?> builder) {
+    this.name = builder.name;
+    this.id = builder.id;
+    this.description = builder.description;
+    this.properties = new HashMap<>(builder.properties);
+    this.updateTime = builder.updateTime;
+    this.contactRelations = List.copyOf(builder.contactRelations);
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public ItemIdentifier getId() {
+    return id;
+  }
+
+  public Optional<String> getDescription() {
+    return Optional.ofNullable(description);
+  }
+
+  public Map<String, PropertyValue> getProperties() {
+    return Collections.unmodifiableMap(properties);
+  }
+
+  public Optional<Instant> getUpdateTime() {
+    return Optional.ofNullable(updateTime);
+  }
+
+  public Collection<ContactRelation> getContactRelations() {
+    return contactRelations;
+  }
+
+  public abstract static class Builder<T, SELF extends Builder<T, ?>> {
```

Comment:
Une meilleure définition serait
```
public abstract static class Builder<T, SELF extends Builder<T, SELF>>
```
ou plutôt 
```
public abstract static class Builder<T, THIS extends Builder<T, THIS>>
```
Par qu'en java self ne signifie rien.

Reply:
Bonne remarque, j'ai modifié le `SELF` et en `THIS`.